### PR TITLE
Support parsing headers that start with `v`

### LINKED ISSUE #352: Headers that start with `v` can't be parsed
- Workflow file: https://github.com/GradleUp/shadow/commit/a2cad8a77bf9dee02435a1326f5f7fe91ce3457b.
- Changelog file: https://github.com/GradleUp/shadow/blob/a2cad8a77bf9dee02435a1326f5f7fe91ce3457b/src/docs/changes/README.md.
- Running result: https://github.com/GradleUp/shadow/actions/runs/10299485780/job/28506855965.

You can repro this issue at https://github.com/GradleUp/shadow/pull/936.

### PR DESCRIPTION
**Pull request by Goooler**
Closes #352.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ async function extractReleaseNotes(changelogFile, prerelease) {
     const lines = []
     let inside_release = false
     for await (const line of rl) {
-        const start_of_release = (!!line.match("^#+ \\[[0-9]") || (prerelease === 'true' && !!line.match("^#+ \\[Unreleased\\]")))
+        const start_of_release = (!!line.match("^#+ \\[[vV]?[0-9]") || (prerelease === 'true' && !!line.match("^#+ \\[Unreleased\\]")))
```
Comment (Goooler) [inline src/index.js:39]: I'm not a JS expert.

This works on my local but CI,

```js
const pattern = /^#+ \[[vV]?[0-9]/;
const str = "## [v8.3.0]";

console.log(pattern.test(str)); // output: true
```

Comment (Goooler) [reply src/index.js:39]: 69a05474cfcb205f998ed8e094972870787f632e

**Review by ffurrer2 — APPROVED** [review]
LGTM

**Comment by Goooler** [comment]
Any plan to publish a new release for this?

**Comment by Goooler** [comment]
Thanks!

**Pull request merged**